### PR TITLE
Managing Docker Containers using the Command Line Interface

### LINKED ISSUE #2891: Managing Docker Containers using Command Line Interface(CLI)
### Proposed title of article
Managing Docker Containers using Command Line Interface

### Introduction paragraph (2-3 paragraphs):
If you are managing numerous Docker Containers simultaneously, you might have noticed that this is indeed a very hefty task. In this article, we will learn some top ways that can be very handy for you to manage Docker Containers straight from the Command Line Interface. We will see practical examples of the most commonly used yet very effective and efficient Docker Commands that would surely make your Docker experience seamless.

### Key takeaways: 
By the end of this article the reader will be able to manage docker containers by:
1. Running a Docker Container in Interactive Shell.
2. Remove all the Dangling Volumes
3. Removing Docker Containers and Images
4. Using Aliases.
5. Inspecting Docker Containers .

### References: 
N/A

### PR DESCRIPTION
**Pull request by Wanyoike-shift**
Managing Docker Containers using Command Line Interface

### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2891

### CODE REVIEW

**Comment by Wanyoike-shift** [comment]
Hello @Bayler I have created another PR as requested. Thank you

**Comment by Bayler** [comment]
@Wanyoike-shift You've forgotten to add your author bio which you had initially added in your previous PR

**Comment by Wanyoike-shift** [comment]
I have added it @Bayler

**Comment by Bayler** [comment]
> I have added it @Bayler

Noted @Wanyoike-shift Are you on Slack? I want to send you an example of a good hero image.

**Comment by Wanyoike-shift** [comment]
yes I am in slack. @Bayler

**Comment by Bayler** [comment]
> yes I am in slack. @Bayler

Let me send you an example of a good hero image that I've helped create for you. Hero images needs to look professional and should capture the eyes of the reader. I personally use {Canva](www.canva.com) to create my hero images. It provides access to free images and templates you can use to create stunning hero images.

**Comment by Wanyoike-shift** [comment]
Hello @Bayler what is the progress of the article.

**Comment by Bayler** [comment]
I'll be finishing the review today @Wanyoike-shift 

But I'll still have to wait for the plagiarism check before I move to final review

File: content/articles/managing-docker-containers-using-command-line-interface/index.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+### Introduction
+
+With Docker, you can build, test, and deploy applications in the form of portable containers that can be used nearly anywhere. When using Docker, you might quickly get a huge number of useless items, which take up a lot of disk space and clog up the Docker command output. Unless you specifically instruct Docker too, it will not delete unneeded items like containers, images, volumes, and networks. Some of the most effective techniques to manage Docker Containers directly from the Command Line Interface are discussed in this article. We will look at some real-world examples of the most often used but very effective and efficient Docker commands that will make your Docker system organized and save up disk space by deleting unneeded Docker containers, images, volumes, and networks.
```
Comment (Bayler) [inline content/articles/managing-docker-containers-using-command-line-interface/index.md]: This sentence, "We will look........." is too long. Consider breaking it into two parts.

File: content/articles/managing-docker-containers-using-command-line-interface/index.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+- [Using interactive shell to run a docker container](#using-interactive-shell-to-run-a-docker-container)
+- [Remove all the dangling volumes](#remove-all-the-dangling-volumes)
+- [Removing docker containers and images](#removing-docker-containers-and-images)
+- [Using aliases](#using-aliases)
+- [Examining Docker containers](#examining-docker-containers)
+- [Conclusion](#conclusion)
+
+### Using interactive shell to run a docker container
+
+In most circumstances, an `interactive shell` reads and writes to the user's terminal. Interactive behavior is enabled when the bash command is used without any non-option arguments, unless the option is a text to read from or the shell is started to read from standard input, in which case positional parameters can be provided.
```
Comment (Bayler) [inline content/articles/managing-docker-containers-using-command-line-interface/index.md:16]: This sentence, "Interactive behavior........." is too long. Consider breaking it into two parts.

File: content/articles/managing-docker-containers-using-command-line-interface/index.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+Assume you have used the following command to get an Ubuntu image from Docker Hub:
+
+```bash
+sudo docker pull ubuntu
+```
+
+Now you want to use an interactive shell to execute the Ubuntu container. After downloading a Docker Ubuntu image from the official `Docker registry`, you will utilize bash on the Ubuntu OS to edit or install packages. The Docker registry is a platform that hosts images. You may do so by using the `-I` flag to execute the Docker Container in interactive mode.
```
Comment (Bayler) [inline content/articles/managing-docker-containers-using-command-line-interface/index.md]: Please provide a link to where a reader can download the Docker Ubuntu image from its official registry.

File: content/articles/managing-docker-containers-using-command-line-interface/index.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+```bash
+alias dockrm='docker rm'
+alias docklist='docker ps -a'
+```
+
+### Examining Docker containers 
+
+The Docker inspect command may be used to acquire information about a specific docker container. It provides you with all of the information about the container, including the path, creation date, status, driver, and so on. The container name is required to check the container.
+
```
Comment (Bayler) [inline content/articles/managing-docker-containers-using-command-line-interface/index.md:90]: Please show us how the docker `inspect` command is used. Don't just state it. This is beneficial to the reader. Also, explain the commands in your article where necessary.

**Comment by Bayler** [comment]
Hi @Wanyoike-shift Welcome again to our Section Engineering program. 

I'm done with the review. Please make the necessary requested changes then we can move to final review. 

Extra: After you've shown us the commands used for a particular task, it's good practice to explain a little bit on it.

**Comment by hectorkambow** [comment]
@Wanyoike-shift 
Upon a quick glance this tutorial/article seems a bit light on substance. 

Just seem rather short - I have not dived into yet.

cc @Bayler

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by Bayler** [comment]
> @Wanyoike-shift
> Upon a quick glance this tutorial/article seems a bit light on substance.
> 
> Just seem rather short - I have not dived into yet.
> 
> cc @Bayler

Yes @hectorkambow I noticed that too. That's why I commented and stated that he should explain a little bit more especially in places where commands have been written. I felt like those bits weren't explained well enough. But I'll follow it up with @Wanyoike-shift

**Comment by Wanyoike-shift** [comment]
hello @Bayler I have done the requested changes.
Please check

**Comment by Bayler** [comment]
> hello @Bayler I have done the requested changes.
> Please check

@Wanyoike-shift You still haven't shown me how the `inspect` command is implemented in code as requested. As I mentioned earlier, it's not enough to state that the `inspect` command is used to acquire.....show us in code how this is done. 

Or did you mean the docker `container` command? Was it a typo? Please correct where necessary.

File: content/articles/managing-docker-containers-using-command-line-interface/index.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+There are times when we must repeat the same command-line instruction, which is made more difficult if the command in question contains parameters, and much more so if we must change the output to make it legible. Most of us end up making some sort of cheat sheet to keep track of all those complicated commands in case we need to execute them again.
+
+We may save time and effort by utilizing an `alias` command to avoid typing or duplicating the same command over and over again. Aliases allow us to execute a command or a group of instructions using a pre-defined `string` that we can tailor to our liking.
+
+To handle aliases, simply create and use aliases in your `/.bashrc` file and mention them.
+
+```bash
+alias dockrm='docker rm'
+alias docklist='docker ps -a'
+```
```
Comment (Bayler) [inline content/articles/managing-docker-containers-using-command-line-interface/index.md:88]: Also, briefly explain these few lines of code. What does `ps` and `-a` represent? Etcetra.

File: content/articles/managing-docker-containers-using-command-line-interface/index.md (hunk 4 above)
Comment (Bayler) [reply content/articles/managing-docker-containers-using-command-line-interface/index.md:93]: The Docker inspect command.....sentence is here. Was it a typo?

**Comment by Wanyoike-shift** [comment]
@Bayler done correcting. thank you

**Comment by Bayler** [comment]
> @Bayler done correcting. thank you

Noted.

**Comment by hectorkambow** [comment]
overall great work - very easy to follow along. 

👍

**Pull request merged**